protocol/proofs/withdrawal-invalidation-mitigation.md

```diff
@@ -0,0 +1,83 @@
+
+Withdrawal invalidation is a powerful tool in the incident response playbook as it can mitigate the
+impact of a large number of invalid dispute games at the same time. However, the it also forces
+users to resubmit their withdrawal proofs and wait an additional 7 days to execute a withdrawal.
+Users have reported that this invalidation and additional delay period is a pain point. This user
+impact means that the fallback is highly disruptive as a standard security mechanism.
+
+We are in the process of further assessing the impact and severity of this pain point but are
+creating a design document in the meanwhile to determine the lift to resolve this issue.
+
+## Proposed Solution
+
+We propose solving this problem by making withdrawal invalidation an optional part of the fallback
+mechanism. We can achieve this by slightly modifying the rules for withdrawal verification.
+
+We currently enforce the following set of rules (not including other rules like blacklist checks):
+
+- Withdrawals must be proven against a dispute game of the `respectedGameType`.
+- Withdrawals must be finalized against a dispute game of the `respectedGameType`.
+- Withdrawals must be finalized against a dispute game created after `respectedGameTypeUpdatedAt`.
+
+We propose removing the requirement that withdrawals must be finalized against a dispute game of
+the `respectedGameType`. Withdrawals that have been previously proven against a dispute game when
+that game *was* respected can be finalized even if that dispute game is no longer of the respected
+game type.
+
+We additionally propose that the `respectedGameTypeUpdatedAt` timestamp not be updated by default.
+Instead, the timestamp can be selectively updated if withdrawal invalidation is actually required.
+Combined with the first proposed change, this proposal means that older withdrawal proofs can
+usually be finalized when the respected game type is changed but withdrawals can still be blanket
+invalidated if necessary.
+
+Finally we also propose to add the following to the set of verification rules:
+
+- Withdrawals must be proven against a dispute game created after `respectedGameTypeUpdatedAt`.
+
+This would have the effect of preventing users from being able to create withdrawal proofs that
+could never actually be used to finalize a withdrawal. Although this is not strictly necessary, it
+blocks off a footgun and we typically like to block off footguns where possible.
+
+### Naming
+
+We may want to consider renaming `respectedGameTypeUpdatedAt` to something more representative of
+the new system given that the variable would not be updating every time that the
+`respectedGameType` variable is changed. Alternatively, we could keep `respectedGameTypeUpdatedAt`
+untouched and instead introduce a new variable that more accurately reflects the system. This would
+also avoid a breaking change to the contract ABI and the semantics of the original variable.
```

Comment:
this variable could be decoupled from `respectedGameType`. Instead it could be called something like `gameCreationValidFrom`, and then the check would be like

```solidity
require(
    createdAt >= gameCreationValidFrom,
    "OptimismPortal: dispute game created before latest creation validity timestamp"
);
```

what do you think?

the occurrences of `respectedGameTypeUpdatedAt` in the monorepo are low. sorry for the noob question but are there many concerns outside of the monorepo? perhaps integrators would be happy for the breaking change here as their withdrawal validity logic should be updated.

Reply:
Realistically I think you're correct here -- it's possible we could even keep the semantic meaning of `respectedGameTypeUpdatedAt` (as in, we do continue to update it when `setRespectedGameType` is called) but add this new variable `gameCreationValidFrom` with these new semantics.